kartothek/io_components/write.py

```diff
@@ -97,49 +88,39 @@ def persist_indices(
     return output_filenames
 
 
-def persist_common_metadata(partition_list, update_dataset, store, dataset_uuid):
-    # hash the schemas for quick equality check with possible false negatives
-    # (e.g. other pandas version or null schemas)
-    tm_dct = defaultdict(set)
-    for mp in partition_list:
-        for tab, tm in mp.table_meta.items():
-            tm_dct[tab].add(tm)
+def persist_common_metadata(
+    schemas: Iterable[SchemaWrapper],
+    update_dataset: Optional[DatasetFactory],
+    store: KeyValueStore,
+    dataset_uuid: str,
+    table_name: str,
+):
+
+    if not schemas:
+        return None
+    schemas_set = set(schemas)
+    del schemas
```

Comment:
Why this delete?

Reply:
mypy doesn't like to assign the _set_ schemas to a variable which is already typed as a _list_. I introduced another variable and are deleting the old one to ensure it is not used by mistake in the code below. at least in my experience this is a fairly common issue with these kinds of refactoring